### PR TITLE
feat: show message box on close of feedback form

### LINKED ISSUE #9192: clicking outside of the feedback form should not close the form
### Bug description

if we open the feedback form and start to fill it then click elsewhere, everything is lost

it should at least notice the user that it's gonna disappear/be closed

### Operating system

macOS

### Installation Method

None

### Version

1.11.1

### Steps to reproduce

_No response_

### Relevant log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by gastoner**
### What does this PR do?
Shows message box on close of feedback form

### Screenshot / video of UI
Screenshot 2024-11-19 132821

### What issues does this PR fix or reference?
Closes #9192

### CODE REVIEW

**Review by benoitf — COMMENTED** [review]
I don't know if we can be a little bit smarter like if I didn't changed anything in the document it's ok to close it

(like if I click on the icon just to see the content and click outside I've not the extra popup)

File: packages/renderer/src/lib/feedback/SendFeedback.svelte [hunk 1]
```diff
@@ -9,6 +9,7 @@ import GitHubIssueFeedback from './feedbackForms/GitHubIssueFeedback.svelte';
 let displayModal = false;
 const DEFAULT_CATEGORY: FeedbackCategory = 'developers';
 let category: FeedbackCategory = DEFAULT_CATEGORY;
+let allEmptyOrZero: boolean = true;
```
Comment (axel7083) [inline packages/renderer/src/lib/feedback/SendFeedback.svelte]: Weird wording ? some ideas `edited`, `hasContent`

File: packages/renderer/src/lib/feedback/SendFeedback.svelte [hunk 2]
```diff
@@ -44,9 +69,10 @@ function hideModal(): void {
     </div>
 
     {#if category === 'developers'}
-      <DevelopersFeedback onCloseForm={hideModal}/>
+      <DevelopersFeedback onCloseForm={hideModal} on:propsAreEmpty={handleUpdate}/>
```
Comment (axel7083) [inline packages/renderer/src/lib/feedback/SendFeedback.svelte]: I would no use the `props` wording, might just provide a `onchange` method, so we keep with _traditional_ props, we can just ignore the content, just capture the event, and set the `allEmptyOrZero` to true or false

File: packages/renderer/src/lib/feedback/feedbackForms/DevelopersFeedback.svelte [hunk 3]
```diff
@@ -9,13 +9,16 @@ import {
...
 import { Button, ErrorMessage, Link } from '@podman-desktop/ui-svelte';
+import { createEventDispatcher } from 'svelte';
 import Fa from 'svelte-fa';
 
 import FeedbackForm from '/@/lib/feedback/FeedbackForm.svelte';
 import type { FeedbackProperties } from '/@api/feedback';
 
 import WarningMessage from '../../ui/WarningMessage.svelte';
 
+const dispatch = createEventDispatcher<{ propsAreEmpty: boolean }>();
```
Comment (axel7083) [inline packages/renderer/src/lib/feedback/feedbackForms/DevelopersFeedback.svelte]: please avoid svelte 4 createEventDispatcher

File: packages/renderer/src/lib/feedback/feedbackForms/DevelopersFeedback.svelte [hunk 4]
```diff
@@ -25,6 +28,8 @@ $: hasFeedback =
   (tellUsWhyFeedback && tellUsWhyFeedback.trim().length > 4) ||
   (contactInformation && contactInformation.trim().length > 4);
 
+$: dispatch('propsAreEmpty', !(smileyRating || tellUsWhyFeedback || contactInformation));
```
Comment (axel7083) [inline packages/renderer/src/lib/feedback/feedbackForms/DevelopersFeedback.svelte]: avoid using dispatch, either migrate the component to svelte 5 or use a callback props

File: packages/renderer/src/lib/feedback/feedbackForms/GitHubIssueFeedback.svelte [hunk 5]
```diff
@@ -1,8 +1,11 @@
 <script lang="ts">
 import { Button, ErrorMessage, Link } from '@podman-desktop/ui-svelte';
+import { createEventDispatcher } from 'svelte';
 
 import FeedbackForm from '/@/lib/feedback/FeedbackForm.svelte';
 
+const dispatch = createEventDispatcher<{ propsAreEmpty: boolean }>();
```
Comment (axel7083) [inline packages/renderer/src/lib/feedback/feedbackForms/GitHubIssueFeedback.svelte]: same remark

**Review by axel7083 — COMMENTED** [review]
I love the feature, it makes totally sense to have this warning, thanks @gastoner !

> I noticed that if you type then delete, you still have the warning dialog, wondering if that matters, or we could handle that 🤔 (if not easily don't)

**Comment by gastoner** [comment]
Migration of forms in https://github.com/podman-desktop/podman-desktop/pull/10000 ! What a PR number 🥇💯

**Review by jeffmaury — COMMENTED** [review]
Feedback Form => Feedbck form ?

**Review by axel7083 — APPROVED** [review]

File: packages/renderer/src/lib/feedback/SendFeedback.svelte [hunk 6]
```diff
@@ -20,15 +21,39 @@ window.events?.receive('display-feedback', () => {
...
+
+async function hideModal(): Promise<void> {
+  // If all of the form fields are empty/ in default state dont show the dialog
+  if (!hasContent) {
+    closeModal();
+    return;
+  }
+
+  const result = await window.showMessageBox({
+    title: 'Close Feedback Form',
```
Comment (jeffmaury) [inline packages/renderer/src/lib/feedback/SendFeedback.svelte]:
```suggestion
    title: 'Close Feedback form',
```

**Review by jeffmaury — APPROVED** [review]
LGTM

**Pull request merged**